Clarify insecure deployment docs

jseldess opened the pull request:
- Update local cluster examples to use `--host=localhost`
- Update cockroach command docs and insecure manual deployment docs

Fixes #1352

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/docs/1358)
<!-- Reviewable:end -->

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/88301102c74c7185f55af630f195ba0e46d3b531/

a-robinson commented:
Maybe I'm wrong, but I'm not really a fan of anything other than the changes to `cockroach start`, especially when we're adding made up hostnames just to indicate that the flag exists. I'd prefer to only change the `cockroach start commands` and keep all the client command examples the same.

---

Reviewed 15 of 15 files at r1.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[build-a-clojure-app-with-cockroachdb.md, line 28 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1358#-KjF05EitWjDVr2a3iPp:-KjF05EitWjDVr2a3iPq:b9qn9to) ([raw file](https://github.com/cockroachdb/docs/blob/88301102c74c7185f55af630f195ba0e46d3b531/build-a-clojure-app-with-cockroachdb.md#L28)):*
> ```Markdown
> ~~~ shell
> $ cockroach sql --insecure \
> --host=localhost \
> ```

`cockroach sql` defaults to connecting to localhost. Why should we explicitly specify it?

---

*[configure-replication-zones.md, line 157 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1358#-KjEzt4vFFJ-D5iwfr6I:-KjEzt4vFFJ-D5iwfr6J:b1corxm) ([raw file](https://github.com/cockroachdb/docs/blob/88301102c74c7185f55af630f195ba0e46d3b531/configure-replication-zones.md#L157)):*
> ```Markdown
> ~~~ shell
> $ cockroach zone get .default --insecure \
> --host=roachcluster.com \
> ```

I'm not a big fan of making the commands not copy-pastable. Can we just leave host and port off here?

---

*[start-a-local-cluster-in-docker.md, line 67 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1358#-KjF1bFEaxShVtfGuIJq:-KjF1bFEaxShVtfGuIJr:b-y2zm50) ([raw file](https://github.com/cockroachdb/docs/blob/88301102c74c7185f55af630f195ba0e46d3b531/start-a-local-cluster-in-docker.md#L67)):*
> ```Markdown
> - `-p 26257:26257 -p 8080:8080`: These flags map the default port for inter-node and client-node communication (`26257`) and the default port for HTTP requests to the Admin UI (`8080`) from the container to the host. This enables inter-container communication and makes it possible to call up the Admin UI from a browser.
> - `-v "//c/Users/<username>/cockroach-data/roach1:/cockroach/cockroach-data"`: This flag mounts a host directory as a data volume. This means that data and logs for this node will be stored in `Users/<username>/cockroach-data/roach1` on the host and will persist after the container is stopped or deleted. For more details, see Docker's <a href="https://docs.docker.com/engine/tutorials/dockervolumes/#/mount-a-host-directory-as-a-data-volume">Mount a host directory as a data volume</a> topic.
> - `cockroachdb/cockroach:{{site.data.strings.version}} start --insecure --host=localhost`: The CockroachDB command to [start a node](start-a-node.html) in the container in insecure mode with the node listening only on `localhost`. Default ports are used for internal and client traffic (`26257`) and for HTTP requests from the Admin UI (`8080`).
> ```

I think this breaks the example - the containers rely on being able to talk to each other's hostnames.

---

*[_includes/start_in_docker/mac-linux-steps.md, line 42 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1358#-KjF1pKVmkI4Lr9N3rsN:-KjF1pKVmkI4Lr9N3rsO:b2e2r5e) ([raw file](https://github.com/cockroachdb/docs/blob/88301102c74c7185f55af630f195ba0e46d3b531/_includes/start_in_docker/mac-linux-steps.md#L42)):*
> ```Markdown
> - `-p 26257:26257 -p 8080:8080`: These flags map the default port for inter-node and client-node communication (`26257`) and the default port for HTTP requests to the Admin UI (`8080`) from the container to the host. This enables inter-container communication and makes it possible to call up the Admin UI from a browser.
> - `-v "${PWD}/cockroach-data/roach1:/cockroach/cockroach-data"`: This flag mounts a host directory as a data volume. This means that data and logs for this node will be stored in `${PWD}/cockroach-data/roach1` on the host and will persist after the container is stopped or deleted. For more details, see Docker's <a href="https://docs.docker.com/engine/tutorials/dockervolumes/#/mount-a-host-directory-as-a-data-volume">Mount a host directory as a data volume</a> topic.
> - `cockroachdb/cockroach:{{site.data.strings.version}} start --insecure`: The CockroachDB command to [start a node](start-a-node.html) in the container in insecure mode with the node listening only on `localhost`. Default ports are used for internal and client traffic (`26257`) and for HTTP requests from the Admin UI (`8080`).
> ```

Same issue here, passing `--host=localhost` will break the demo

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1358#-:-KjF1wL6nx_oDoiJLMjE:btocgx0)*
<!-- Sent from Reviewable.io -->

jseldess commented:
I made changes to client commands because I was assuming that people who need to set `--host=localhost` to get the cluster running (e.g., Diana and others) would also have problems with client commands if `--host` weren't set explicitly.  Is that not true?

---

Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[build-a-clojure-app-with-cockroachdb.md, line 28 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1358#-KjF05EitWjDVr2a3iPp:-KjFInFyvsJugXSo5nbN:bhrybzq) ([raw file](https://github.com/cockroachdb/docs/blob/88301102c74c7185f55af630f195ba0e46d3b531/build-a-clojure-app-with-cockroachdb.md#L28)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

`cockroach sql` defaults to connecting to localhost. Why should we explicitly specify it?
</blockquote></details>

I made changes to client commands because I was assuming that people who need to set `--host=localhost` to get the local cluster running (e.g., Diana and others) would also have problems with client commands if `--host` weren't set explicitly.  Is that not true?

---

*[configure-replication-zones.md, line 157 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1358#-KjEzt4vFFJ-D5iwfr6I:-KjFLEqMRKXdn7ywvBc1:bs4ptgk) ([raw file](https://github.com/cockroachdb/docs/blob/88301102c74c7185f55af630f195ba0e46d3b531/configure-replication-zones.md#L157)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

I'm not a big fan of making the commands not copy-pastable. Can we just leave host and port off here?
</blockquote></details>

Given that you'd need to specify the host and port of the server you're connecting to, don't you find it misleading/incomplete not to include these details?

Would you be ok with this if it were all on one line with placeholder text in `--host`?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1358#-:-KjFL31cO2QjHgQy2vJB:b-6h5e8l)*
<!-- Sent from Reviewable.io -->

jseldess commented:
Review status: 13 of 14 files reviewed at latest revision, 4 unresolved discussions.

---

*[start-a-local-cluster-in-docker.md, line 67 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1358#-KjF1bFEaxShVtfGuIJq:-KjFMXtyKYakBqaI6G_r:bs7wumx) ([raw file](https://github.com/cockroachdb/docs/blob/88301102c74c7185f55af630f195ba0e46d3b531/start-a-local-cluster-in-docker.md#L67)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

I think this breaks the example - the containers rely on being able to talk to each other's hostnames.
</blockquote></details>

Done. 

---

*[_includes/start_in_docker/mac-linux-steps.md, line 42 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1358#-KjF1pKVmkI4Lr9N3rsN:-KjFMZXDulhnVtdbgLTo:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/88301102c74c7185f55af630f195ba0e46d3b531/_includes/start_in_docker/mac-linux-steps.md#L42)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

Same issue here, passing `--host=localhost` will break the demo
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1358)*
<!-- Sent from Reviewable.io -->

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/a3efc405597f77ba344d213435678a6793dc7207/

a-robinson commented:
Nope, setting `--host=localhost` isn't necessary for client commands because `localhost` is the default value for the `--host` flag for all client commands.

---

Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[build-a-clojure-app-with-cockroachdb.md, line 28 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1358#-KjF05EitWjDVr2a3iPp:-KjIz_uuQ72jqRwNUUXv:b-yff2gg) ([raw file](https://github.com/cockroachdb/docs/blob/88301102c74c7185f55af630f195ba0e46d3b531/build-a-clojure-app-with-cockroachdb.md#L28)):*
<details><summary><i>Previously, jseldess wrote…</i></summary><blockquote>

I made changes to client commands because I was assuming that people who need to set `--host=localhost` to get the local cluster running (e.g., Diana and others) would also have problems with client commands if `--host` weren't set explicitly.  Is that not true?
</blockquote></details>

As mentioned above, all client commands default to using `localhost`.

---

*[configure-replication-zones.md, line 157 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1358#-KjEzt4vFFJ-D5iwfr6I:-KjIzlRTHCcxmdsJhtj9:b-kt6fru) ([raw file](https://github.com/cockroachdb/docs/blob/88301102c74c7185f55af630f195ba0e46d3b531/configure-replication-zones.md#L157)):*
<details><summary><i>Previously, jseldess wrote…</i></summary><blockquote>

Given that you'd need to specify the host and port of the server you're connecting to, don't you find it misleading/incomplete not to include these details?

Would you be ok with this if it were all on one line with placeholder text in `--host`?
</blockquote></details>

You're the one with the creative authority here, so I'm ok with whatever you think is most appropriate :) It's just my opinion that unless we showed how to start a server that doesn't listen on localhost and the default port somewhere earlier in the page, I'd assume that the default flags are good enough. Anyone who has configured their server to use non-default values should be more able to figure out that they need to do so with the client, while people that have only ever used default server values might be somewhat confused by these flags.

---

*[start-a-local-cluster-in-docker.md, line 67 at r2](https://reviewable.io:443/reviews/cockroachdb/docs/1358#-KjJ-ZaZP1eGHKsqZuSh:-KjJ-ZaZP1eGHKsqZuSi:bjzzwmy) ([raw file](https://github.com/cockroachdb/docs/blob/a3efc405597f77ba344d213435678a6793dc7207/start-a-local-cluster-in-docker.md#L67)):*
> ```Markdown
> - `-p 26257:26257 -p 8080:8080`: These flags map the default port for inter-node and client-node communication (`26257`) and the default port for HTTP requests to the Admin UI (`8080`) from the container to the host. This enables inter-container communication and makes it possible to call up the Admin UI from a browser.
> - `-v "//c/Users/<username>/cockroach-data/roach1:/cockroach/cockroach-data"`: This flag mounts a host directory as a data volume. This means that data and logs for this node will be stored in `Users/<username>/cockroach-data/roach1` on the host and will persist after the container is stopped or deleted. For more details, see Docker's <a href="https://docs.docker.com/engine/tutorials/dockervolumes/#/mount-a-host-directory-as-a-data-volume">Mount a host directory as a data volume</a> topic.
> - `cockroachdb/cockroach:{{site.data.strings.version}} start --insecure --host=localhost`: The CockroachDB command to [start a node](start-a-node.html) in the container in insecure mode.
> ```

It looks like you missed this `--host=localhost` bit

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1358#-:-KjIzNAaWQLiDvnSXqv_:batnm3q)*
<!-- Sent from Reviewable.io -->

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/92f31961c645e7e79528c957b8523ac690f6a3af/

jseldess commented:
Review status: 3 of 14 files reviewed at latest revision, 5 unresolved discussions.

---

*[configure-replication-zones.md, line 157 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1358#-KjEzt4vFFJ-D5iwfr6I:-KjJHypc6xD-0HKpbDq-:b-o3xfv2) ([raw file](https://github.com/cockroachdb/docs/blob/88301102c74c7185f55af630f195ba0e46d3b531/configure-replication-zones.md#L157)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

You're the one with the creative authority here, so I'm ok with whatever you think is most appropriate :) It's just my opinion that unless we showed how to start a server that doesn't listen on localhost and the default port somewhere earlier in the page, I'd assume that the default flags are good enough. Anyone who has configured their server to use non-default values should be more able to figure out that they need to do so with the client, while people that have only ever used default server values might be somewhat confused by these flags.
</blockquote></details>

I added `localhost` as the `--host` default on all relevant client command docs, and I reverted the examples. Eventually, these pages should be updated to show examples for both local and real clusters. TFTR, @a-robinson!

---

*[start-a-local-cluster-in-docker.md, line 67 at r2](https://reviewable.io:443/reviews/cockroachdb/docs/1358#-KjJ-ZaZP1eGHKsqZuSh:-KjJHpVGqnkunC38tDta:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/a3efc405597f77ba344d213435678a6793dc7207/start-a-local-cluster-in-docker.md#L67)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

It looks like you missed this `--host=localhost` bit
</blockquote></details>

Done.

---

*[build-a-clojure-app-with-cockroachdb.md, line 28 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1358#-KjF05EitWjDVr2a3iPp:-KjJHthtGwO133ntIVZe:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/88301102c74c7185f55af630f195ba0e46d3b531/build-a-clojure-app-with-cockroachdb.md#L28)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

As mentioned above, all client commands default to using `localhost`.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1358)*
<!-- Sent from Reviewable.io -->

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/532ef7623c2d2efedbcaedb0df105688418e8838/

a-robinson commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

---

Reviewed 1 of 1 files at r3, 12 of 14 files at r4.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[configure-replication-zones.md, line 339 at r4](https://reviewable.io:443/reviews/cockroachdb/docs/1358#-KjJJN1Gi4p16JuiXLa_:-KjJJN1Gi4p16JuiXLaa:btq17r0) ([raw file](https://github.com/cockroachdb/docs/blob/532ef7623c2d2efedbcaedb0df105688418e8838/configure-replication-zones.md#L339)):*
> ```Markdown
>    ~~~ shell
>    # Apply the replication zone to the database used by application 1:
>    $ cockroach zone set app1_db --insecure --host=<hostname of any node> -f app1_zone.yaml
> ```

Just double checking that keeping these here was intentional?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1358#-:-KjJK0J5BR6YAs7CZjoN:bg7igkv)*
<!-- Sent from Reviewable.io -->

jseldess commented:
Review status: 13 of 14 files reviewed at latest revision, 5 unresolved discussions.

---

*[configure-replication-zones.md, line 339 at r4](https://reviewable.io:443/reviews/cockroachdb/docs/1358#-KjJJN1Gi4p16JuiXLa_:-KjJRDNIO22yPiQHK2j7:b5pqjsi) ([raw file](https://github.com/cockroachdb/docs/blob/532ef7623c2d2efedbcaedb0df105688418e8838/configure-replication-zones.md#L339)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

Just double checking that keeping these here was intentional?
</blockquote></details>

Good catch. I've removed `--host` from these client command steps and clarified that you're creating zone configs on one of the nodes. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1358)*
<!-- Sent from Reviewable.io -->

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/85c4c2105d8dd8b2a3f4092e059398002e961991/
